src/main/java/tuteez/model/person/lesson/Lesson.java

```diff
@@ -168,18 +157,53 @@ public boolean equals(Object other) {
         }
 
         Lesson otherLesson = (Lesson) other;
-        return dayAndTime.equals(((Lesson) other).dayAndTime);
+        return otherLesson.startTime.equals(this.startTime) && otherLesson.endTime.equals(this.endTime);
     }
 
     @Override
     public int hashCode() {
-        return dayAndTime.hashCode();
+        return Objects.hash(lessonDay, startTime, endTime);
     }
 
     /**
      * Format state as text for viewing.
      */
     public String toString() {
-        return '[' + dayAndTime + ']';
+        return '[' + lessonDay.toString() + " " + startTime.toString().replace(":", "") + "-"
+                + endTime.toString().replace(":", "") + ']';
+    }
+
+    /**
+     * @return A string with only day and time
+     */
+    public String dayTimeString() {
```

Comment:
Perhaps this can be named `getDayAndTime()` instead, to follow method naming conventions (start with verb)

Reply:
Okay Ill change that in the next PR